lib/action_view/component/base.rb

```diff
@@ -58,6 +60,9 @@ class Base < ActionView::Base
       #
       def render_in(view_context, *args, &block)
         self.class.compile
+        self.controller = view_context.controller
+        @view_renderer ||= view_context.view_renderer
+        @lookup_context ||= view_context.lookup_context
```

Comment:
Can you please add
```
@view_flow ||= view_context.view_flow
```
So that `content_for` works?

Reply:
I've added that now, together with a test.